cmd/gitcloner/cmd/cmd.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type GitCloner interface {
+	CloneRepo(opts *Options) error
+}
+
+type Options struct {
+	Repo              string
+	Path              string
+	Branch            string
+	Revison           string
+	CABundleFile      string
+	Username          string
+	PasswordFile      string
+	SSHPrivateKeyFile string
+	InsecureSkipTLS   bool
+	KnownHostsFile    string
+}
+
+var opts *Options
+
+func New(gitCloner GitCloner) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gitcloner [REPO] [PATH]",
+		Short: "Clones a git repository",
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
You can use `Args` to validate positional arguments:
```suggestion
		Args: cobra.ExactArgs(2),
		RunE: func(cmd *cobra.Command, args []string) error {
			opts.Repo = args[0]
			opts.Path = args[1]
			return gitCloner.CloneRepo(opts)
		}
```
That way you can drop `cloneRepoWithArgs`.

Reply:
nice! I didn't know about cobra.ExactArgs() https://github.com/rancher/gitjob/pull/302/commits/1a5b08e28135e6d71bbb2c4d54c8228efb39c75b